src/sentry/integrations/source_code_management/commit_context.py

```diff
@@ -292,88 +492,669 @@ def queue_comment_task_if_needed(
+                issue_ids=issue_id_list,
+                comment_type=CommentType.OPEN_PR,
+                metrics_base=OPEN_PR_METRICS_BASE,
+                language=language,
+            )
+        except ApiError as e:
+            if self.on_create_or_update_comment_error(
+                api_error=e, metrics_base=OPEN_PR_METRICS_BASE
+            ):
+                return
+
+            metrics.incr(
+                OPEN_PR_METRICS_BASE.format(integration=self.integration_name, key="error"),
+                tags={"type": "api_error"},
             )
+            raise
+
+
+def run_pr_comment_workflow(integration_name: str, pullrequest_id: int, project_id: int) -> None:
+    cache_key = debounce_pr_comment_cache_key(pullrequest_id=pullrequest_id)
+
+    try:
+        pr = PullRequest.objects.get(id=pullrequest_id)
+        assert isinstance(pr, PullRequest)
+    except PullRequest.DoesNotExist:
+        cache.delete(cache_key)
+        logger.info(_pr_comment_log(integration_name=integration_name, suffix="pr_missing"))
+        return
+
+    try:
+        organization = Organization.objects.get_from_cache(id=pr.organization_id)
+        assert isinstance(organization, Organization)
+    except Organization.DoesNotExist:
+        cache.delete(cache_key)
+        logger.info(_pr_comment_log(integration_name=integration_name, suffix="org_missing"))
+        metrics.incr(
+            MERGED_PR_METRICS_BASE.format(integration=integration_name, key="error"),
+            tags={"type": "missing_org"},
+        )
+        return
+
+    try:
+        repo = Repository.objects.get(id=pr.repository_id)
+        assert isinstance(repo, Repository)
+    except Repository.DoesNotExist:
+        cache.delete(cache_key)
+        logger.info(
+            _pr_comment_log(integration_name=integration_name, suffix="repo_missing"),
+            extra={"organization_id": organization.id},
+        )
+        metrics.incr(
+            MERGED_PR_METRICS_BASE.format(integration=integration_name, key="error"),
+            tags={"type": "missing_repo"},
+        )
+        return
+
+    integration = integration_service.get_integration(
+        integration_id=repo.integration_id, status=ObjectStatus.ACTIVE
+    )
+    if not integration:
+        cache.delete(cache_key)
+        logger.info(
+            _pr_comment_log(integration_name=integration_name, suffix="integration_missing"),
+            extra={"organization_id": organization.id},
+        )
+        metrics.incr(
+            MERGED_PR_METRICS_BASE.format(integration=integration_name, key="error"),
+            tags={"type": "missing_integration"},
+        )
+        return
+
+    installation = integration.get_installation(organization_id=organization.id)
+    assert isinstance(installation, CommitContextIntegration)
+
+    installation.run_pr_comment_workflow(
+        organization=organization,
+        repo=repo,
+        pr=pr,
+        project_id=project_id,
+    )
```

Comment:
i think it's okay to move this into the original task instead of moving it into a function and calling it anyway